itests/self_sent_txn_test.go

```diff
@@ -0,0 +1,58 @@
+package itests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/filecoin-project/go-state-types/network"
+
+	"github.com/filecoin-project/go-state-types/big"
+	"github.com/filecoin-project/go-state-types/exitcode"
+	"github.com/filecoin-project/lotus/api"
+	"github.com/filecoin-project/lotus/chain/types"
+	"github.com/filecoin-project/lotus/itests/kit"
+	"github.com/stretchr/testify/require"
+)
+
+func TestSelfSentTxn(t *testing.T) {
```

Comment:
For completeness, it'd be nice to add the exact same test for v14, and assert that both messages succeed there.